src/coprocessor/dag/builder.rs

```diff
@@ -139,13 +140,22 @@ impl DAGBuilder {
             }
         }
 
-        // Currently we only support table scan executor. So if there are more
-        // executors, it is unexpected.
-        if executor_descriptors.next().is_some() {
-            return Err(Error::Other(box_err!(
-                "Unexpected non-first executor {:?}",
-                first_ed.get_tp()
-            )));
+        for ex in executor_descriptors {
+            match ex.get_tp() {
+                ExecType::TypeLimit => {
+                    executor = Box::new(BatchLimitExecutor::new(
+                        executor,
+                        ex.get_limit().get_limit() as usize,
```

Comment:
It looks wired. How about changing one of them?

Reply:
It's a type from mod `tipb`